gitlabform/processors/multiple_entities_processor.py

```diff
@@ -16,22 +16,52 @@ def __init__(
         self,
         configuration_name: str,
         gitlab: GitLab,
-        list_method_name: str,
-        add_method_name: str,
-        delete_method_name: str,
+        # TODO: Should the option of passing as str be deprecated ? It makes life harder (for Mypy and for the devs),
+        #    and it's not like the method is not known|doesn't exist at compile-time
+        list_method_name: Union[str, Callable[[str], list]],
+        add_method_name: Union[str, Callable[[str, Any], None]],
+        delete_method_name: Union[str, Callable[[str, dict], None]],
         defining: AbstractKey,
         required_to_create_or_update: AbstractKey,
-        edit_method_name=None,
+        edit_method_name: Union[str, Callable[[str, str, Any], None], None] = None,
     ):
+        """
+
+        :param configuration_name: The key in the YAML cfg file that determines the start of this Entity's section
+        :param gitlab:
+        :param list_method_name:
+        :param add_method_name:
+        :param delete_method_name:
+        :param defining: TODO -> what is this for ?
```

Comment:
This field contains an expression that tells which fields of the entity in GitLab are _identifying_ it.

It's used for finding matches between entities in your config and the ones in GitLab.

Usually it's just the a single field like "name", "key" or "title", but sometimes it's more complicated.

An example of that case is GroupLDAPLinksProcess where we have `And(Key("provider"), Or(Key("cn"), Key("filter")))` that can be interpreted as "if you find group LDAP link in config and in GitLab that have the same value of field "provider" and the same value of at least one of the fields "cn" and "filter", then they corresponding".

If it's still not clear, please let me know. :)